### PR TITLE
feat: add /init command to generate and auto-load project context

### PR DESCRIPTION
**Pull request by hongkongkiwi**
- Add /init command that generates .groq/context.md and .groq/context.json containing repository tree, languages, config files, and summary
- Auto-load project context at agent startup as system message when available
- Support environment overrides (GROQ_CONTEXT_FILE, GROQ_CONTEXT_DIR)
- Add comprehensive test suite using AVA for context generation and loading
- Include npm test script for easier test execution
- Apply robustness improvements for error handling and edge cases

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,8 +10,9 @@
 		"node": ">=16"
 	},
 	"scripts": {
-		"build": "tsc",
-		"dev": "tsc --watch"
+    "build": "tsc",
+    "dev": "tsc --watch",
+    "test": "ava"
```
Comment (benank) [inline package.json:15]:
```suggestion
		"build": "tsc",
		"dev": "tsc --watch",
		"test": "ava"
```

Comment (benank) [reply package.json:15]: Let's keep the indentation the same

**Review by benank — CHANGES_REQUESTED** [review]
Nice work! Just one quick change and we can get this merged!

**Review by benank — APPROVED** [review]
